### PR TITLE
feat(cacheClient): add option to disable auth

### PR DESCRIPTION
**Pull request by dwojno**

### CODE REVIEW

File: src/cacheServerClient/cacheServerClient.ts [hunk 1]
```diff
@@ -10,15 +10,23 @@ export class CacheServerClient implements ICacheServerClient {
...
   private failedRequests: Array<() => void> = [];
+  private authEnabled: boolean;
 
-  constructor({ url }: { url: string }) {
+  constructor({
+    url,
+    cacheServerSupportsAuth = true
+  }: {
+    url: string;
+    cacheServerSupportsAuth?: boolean;
```
Comment (jrhender) [inline src/cacheServerClient/cacheServerClient.ts:20]: @dwojno How does the consumer (e.g. Switchboard app) of iam-client-lib set this `cacheServerSupportsAuth` param?

Comment (dwojno) [reply src/cacheServerClient/cacheServerClient.ts:20]: Like this:
```
        const cacheClient = new CacheServerClient({ url: 'http://cache-server.com', cacheServerSupportsAuth: false })

        this._iam = new IAM({
          rpcUrl: 'https://volta-rpc.energyweb.org/',
          chainId: 73799,
          cacheClient // optional
        });
```

Comment (jrhender) [reply src/cacheServerClient/cacheServerClient.ts:20]: Ah right, I forgot they pass in the cache server client.

**Review by jrhender — APPROVED** [review]
👍

**Pull request merged**